webui/src/lib/onboarding/repoOnboardingService.tsx

```diff
@@ -0,0 +1,136 @@
+  return `sample_${randomString}.yaml`;
+};
+
+export const isSampleRepo = async (
+  repo: string,
+  defaultBranch: string,
+): Promise<boolean> => {
+  const objectList = await objects.list(repo, defaultBranch, "");
+  return objectList.results.some(
+    // when the API client is typed, we can remove the eslint-disable
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (object: any) => object?.path === SAMPLE_REPO_FILE_NAME
+  );
+};
+
+export const canUseRepoOnboarding = async (
+  repoId: string,
+  defaultBranch: string,
+): Promise<boolean> => {
+  const promises = [
+    retention.setGCPolicyPreflight(repoId),
+    branchProtectionRules.createRulePreflight(repoId),
+    objects.list(
+      repoId,
+      defaultBranch,
+      `${HOOKS_PATH}${generateRandomFileName()}`
+    ),
+  ];
+  
+  try {
+    await Promise.all(promises);
+    return true;
+  } catch (e) {
+    return false;
+  }
+};
+
+export const getRepoOnboardingSteps = (
+  currentRepo: string,
+  defaultBranch: string,
+  objectStoreName: string,
+  navigate: NavigateFunction
+): OnBoardingSteps => [
+  {
+    id: "import-data",
+    title: "Import your existing data",
+    description: (
+      <>
+        <p>Import your existing data from ${objectStoreName} to your repository by creating pointers to it (no copying of data involved).</p>
+        <p>Please see our documentation for information on the <a href="https://docs.lakefs.io/howto/import.html#limitations" target="_blank">limitations of the import process</a> and <a href="https://docs.lakefs.io/howto/import.html#working-with-imported-data" target="_blank">working with imported data</a></p>,
+      </>
+    ),
+    cta: "Run import",
+    onClick: () =>
+      navigate(`/repositories/${currentRepo}/objects?importDialog=true`),
+    showStep: () => true, // objectStoreName !== "local",
+    isCompleted: async () => {
+      const repoBranches = await branches.list(currentRepo);
+      // when the API client is typed, we can remove the eslint-disable
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      return repoBranches.results.some((branch: any) =>
+        IMPORT_BRANCH_REGEX.test(branch.id)
+      );
+    },
+  },
+  {
+    id: "garbage-collection",
+    title: "Setup garbage collection",
+    description: (
+      <>
+        <p>Set up rules to automatically manage when data can be deleted from your repository</p>
+        <p>For more information about setting up and running garbage collection on your lakeFS repository, see the <a href="https://docs.lakefs.io/howto/garbage-collection-committed.html" target="_blank">documentation</a></p>
+      </>
+    ),
+    cta: "Configure",
+    onClick: () => navigate(`/repositories/${currentRepo}/settings/retention`),
+    showStep: () => true,
+    isCompleted: async () => {
+      try {
+        await retention.getGCPolicy(currentRepo);
+        return true;
+      } catch (e) {
+        return false;
```

Comment:
Test for specific types of errors here.  Otherwise e.g a spurious failure to get GC policy will cause this code to nag.

Reply:
Done.